File: app/src/androidTest/java/com/example/finalproject/ViewData.java
package com.example.finalproject;

import android.util.Log;

import com.amplifyframework.core.Amplify;

import android.util.Log;
import com.amplifyframework.core.Amplify;
import com.amplifyframework.datastore.generated.model.Post;


public class ViewData {
    Amplify.DataStore.query(Post.class,
    queryMatches -> {
        if (queryMatches.hasNext()) {
            Log.i("MyAmplifyApp", "Successful query, found posts.");
        } else {
            Log.i("MyAmplifyApp", "Successful query, but no posts.");
        }
    },
    error -> Log.e("MyAmplifyApp",  "Error retrieving posts", error)
            );
}

File: data_pipeline/view_data.dart
import 'package:amplify_flutter/amplify.dart';
import 'package:finalproject/models/Post.dart'; // Import the correct model class
import 'dart:developer' as developer;

class ViewData {
  void queryPosts() {
    Amplify.DataStore.query(
      Post.classType,
      onData: (List<Post> queryMatches, {bool hasNextToken}) {
        if (queryMatches.isNotEmpty) {
          developer.log('Successful query, found posts.', name: 'MyAmplifyApp');
        } else {
          developer.log('Successful query, but no posts.', name: 'MyAmplifyApp');
        }
      },
      onError: (error) =>
          developer.log('Error retrieving posts: $error', name: 'MyAmplifyApp'),
    );
  }
}
